tests/ixia/ecn/test_ecn_markings_cisco_8000.py

```diff
@@ -0,0 +1,161 @@
+        pytest.skip('test_ecn_markings is disabled')
+
+    dut_hostname, dut_port = rand_one_dut_portname_oper_up.split('|')
+    dut_hostname2, lossless_prio = rand_one_dut_lossless_prio.split('|')
+    pytest_require(rand_one_dut_hostname == dut_hostname == dut_hostname2, \
+        'Priority and port are not mapped to the expected DUT')
+
+    testbed_config, port_config_list = ixia_testbed_config
+    duthost = duthosts[rand_one_dut_hostname]
+    lossless_prio = int(lossless_prio)
+
+    setup_ecn_markings_dut(duthost, localhost, ecn_dequeue_marking=dequeue_marking, ecn_latency_marking=latency_marking)
+    stop_debug_shell(duthost)
+
+    duthost.shell('sonic-clear pfccounters')
+    duthost.shell('sonic-clear queuecounters')
+    # TODO: This will be enabled when the PR for get_sai_attributes is approved: https://github.com/Azure/sonic-mgmt/pull/6040
+    #get_sai_attributes(duthost, ptfhost, dut_port, [], clear_only=True)
+    data_traffic_rate = TRAFFIC_RATE / number_of_transmit_ports
+
+    run_ecn_test(api=ixia_api,
+                 testbed_config=testbed_config,
+                 port_config_list=port_config_list,
+        	 conn_data=conn_graph_facts,
+        	 fanout_data=fanout_graph_facts,
+        	 duthost=duthost,
+        	 dut_port=dut_port,
+        	 kmin=kmin,
+        	 kmax=kmax,
+        	 pmax=pmax,
+        	 data_pkt_size=data_pkt_size,
+        	 lossless_prio=lossless_prio,
+        	 prio_dscp_map=prio_dscp_map,
+        	 iters=1,
+        	 xoff_quanta=xoff_quanta,
+        	 data_traffic_rate=data_traffic_rate,
+        	 number_of_transmit_ports=number_of_transmit_ports,
+        	 pfc_pkt_count=1,
+                 enable_capture=False)
+    time.sleep(8)
+
+    sai_values = None
+    total = 'N/A'
+    marked_pkts = 'N/A'
+    # TODO:This will be enabled when the PR for get_sai_attributes is approved: https://github.com/Azure/sonic-mgmt/pull/6040
+    #sai_values = get_sai_attributes(duthost, ptfhost, dut_port, ["SAI_QUEUE_STAT_PACKETS","SAI_QUEUE_STAT_WRED_ECN_MARKED_PACKETS"])
+
+    if sai_values:
+        value_list = eval(sai_values[0])
+        total = value_list[lossless_prio][0]
+        marked_pkts = value_list[lossless_prio][1]
+
+    print ('The SAI attributes in the DUT:{}').format(sai_values)
+    print ('Xoff = {}').format(xoff_quanta)
+    print ('pmax = {}').format(pmax)
+    with open(MARKING_DATA_FILE, 'a') as (fd):
+        fd.write(('{tx_ports},{rate},{deq},{lat},{data_pkt_size},{kmin},{kmax},{pmax},{xoff_quanta},{total},{marked},{time_stamp}\n').format(
+                 time_stamp=str(datetime.datetime.now()),
+        	 rate=data_traffic_rate,
+        	 deq=dequeue_marking,
+        	 lat=latency_marking,
+        	 data_pkt_size=data_pkt_size,
+        	 tx_ports=number_of_transmit_ports,
+        	 kmax=kmax,
+        	 kmin=kmin,
+        	 pmax=pmax,
+        	 xoff_quanta=xoff_quanta,
+        	 total=total,
+        	 marked=marked_pkts))
+    print 'test done'
```

Comment:
It seems that there is no logic to judge if the test passes.

Reply:
The test is passed if run_ecn_test is successfully done. That function itself has the required asserts to check.

Reply:
> The test is passed if run_ecn_test is successfully done. That function itself has the required asserts to check.

Where are the required asserts in run_ecn_test?

Reply:
@baiwei0427 , I was referring to the following checks. If any of them fails, the test will fail.

grep -n assert files/helper.py 
4:from tests.common.helpers.assertions import pytest_assert
89:    pytest_assert(testbed_config is not None, 'Failed to get L2/3 testbed config')
99:    pytest_assert(config_result is True, 'Failed to configure WRED/ECN at the DUT')
108:    pytest_assert(config_result is True, 'Failed to configure PFC threshold to 8')
116:    pytest_assert(port_id is not None,
221:    pytest_assert(len(tx_port_id_list) > 0, "Cannot find any TX ports")
224:    pytest_assert(tx_port_id_list != [], "Cannot find a suitable TX port")
396:    pytest_assert(attempts < max_attempts,